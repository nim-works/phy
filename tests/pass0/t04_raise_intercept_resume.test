discard """
  output: "(UnhandledException 100)"
"""
(TypeDefs
  (ProcTy (Void))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 0) (Locals)
    (Continuations
      (Continuation (Params) 0
        (Raise (IntVal 100) (List (Continue 1) (Unwind)))
      )
      (Subroutine (Params) 0
        (Continue)
      )
      (Continuation (Params))
    )
  )
)