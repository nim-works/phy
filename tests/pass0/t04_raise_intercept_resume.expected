.type t0 (Proc (Void))
.start t0 p0
  LdImmInt 100
  Raise
.eh ((Subroutine L2) End)
.label L2
  Begin
  End
  Ret
.end
